include/odp/api/spec/system_info.h

```diff
@@ -23,6 +24,165 @@ extern "C" {
  *  @{
  */
 
+/**
+ * CPU instruction set architecture (ISA) families
+ */
+typedef enum odp_cpu_arch_t {
+	/** Unknown CPU architecture */
+	ODP_CPU_ARCH_UNKNOWN = 0,
+
+	/** ARM */
+	ODP_CPU_ARCH_ARM,
+
+	/** MIPS */
+	ODP_CPU_ARCH_MIPS,
+
+	/** PowerPC */
+	ODP_CPU_ARCH_PPC,
+
+	/** RISC-V */
+	ODP_CPU_ARCH_RISCV,
+
+	/** x86 */
+	ODP_CPU_ARCH_X86
+
+} odp_cpu_arch_t;
+
+/**
+ * ARM ISA versions
+ *
+ * ISA versions are defined in ascending order.
+ */
+typedef enum odp_cpu_arch_arm_t {
+	/** Unknown ARM ISA version */
+	ODP_CPU_ARCH_ARM_UNKNOWN = 0,
+
+	/** ARMv6 ISA */
+	ODP_CPU_ARCH_ARMV6,
+
+	/** ARMv7-A ISA */
+	ODP_CPU_ARCH_ARMV7,
+
+	/** ARMv8.0-A ISA */
+	ODP_CPU_ARCH_ARMV8_0,
+
+	/** ARMv8.1-A ISA */
+	ODP_CPU_ARCH_ARMV8_1,
+
+	/** ARMv8.2-A ISA */
+	ODP_CPU_ARCH_ARMV8_2,
+
+	/** ARMv8.3-A ISA */
+	ODP_CPU_ARCH_ARMV8_3,
+
+	/** ARMv8.4-A ISA */
+	ODP_CPU_ARCH_ARMV8_4,
+
+	/** ARMv8.5-A ISA */
+	ODP_CPU_ARCH_ARMV8_5,
+
+	/** ARMv8.6-A ISA */
+	ODP_CPU_ARCH_ARMV8_6,
+
+} odp_cpu_arch_arm_t;
+
+/**
+ * MIPS ISA versions
+ */
+typedef enum odp_cpu_arch_mips_t {
+	/** Unknown MIPS ISA version */
+	ODP_CPU_ARCH_MIPS_UNKNOWN = 0
+
+} odp_cpu_arch_mips_t;
+
+/**
+ * PowerPC ISA versions
+ */
+typedef enum odp_cpu_arch_ppc_t {
+	/** Unknown PPC ISA version */
+	ODP_CPU_ARCH_PPC_UNKNOWN = 0
+
+} odp_cpu_arch_ppc_t;
+
+/**
+ * RISC-V ISA versions
+ */
+typedef enum odp_cpu_arch_riscv_t {
+	/** Unknown RISC-V ISA version */
+	ODP_CPU_ARCH_RISCV_UNKNOWN = 0
+
+} odp_cpu_arch_riscv_t;
+
+/**
+ * x86 ISA versions
+ */
+typedef enum odp_cpu_arch_x86_t {
+	/** Unknown x86 ISA version */
+	ODP_CPU_ARCH_X86_UNKNOWN = 0,
+
+	/** x86 32bit ISA */
+	ODP_CPU_ARCH_X86_I686,
+
+	/** x86 64bit ISA */
+	ODP_CPU_ARCH_X86_64
+
+} odp_cpu_arch_x86_t;
+
+/**
+ * CPU ISA versions
+ */
+typedef union odp_cpu_arch_isa_t {
+	/** ARM ISA versions */
+	odp_cpu_arch_arm_t arm;
+
+	/** MIPS ISA versions */
+	odp_cpu_arch_mips_t mips;
+
+	/** PowerPC ISA versions */
+	odp_cpu_arch_ppc_t ppc;
+
+	/** RISC-V ISA versions */
+	odp_cpu_arch_riscv_t riscv;
+
+	/** x86 ISA versions */
+	odp_cpu_arch_x86_t x86;
+
+} odp_cpu_arch_isa_t;
+
+/**
+ * System info
+ */
+typedef struct odp_system_info_t {
+	/**
+	 * CPU architecture
+	 *
+	 * Defines CPU ISA family: ARM, MIPS, PPC, RISC-V or x86
+	 */
+	odp_cpu_arch_t cpu_arch;
+
+	/**
+	 * ISA version of ODP software
+	 *
+	 * Defines the ISA version that was used to build the ODP library. Note that CPU hardware
+	 * may support a higher ISA version than this.
+	 */
+	odp_cpu_arch_isa_t cpu_isa_sw;
```

Comment:
It could be useful to have also `cpu_isa_hw ` variant. `Unknown` would be a valid value for this member since there may not always be a way to know the HW ISA.

Reply:
Added in v7.